HuntTheWumpusLib/Dungeon.cpp

```diff
@@ -157,6 +158,21 @@ namespace HuntTheWumpus
 
     void Dungeon::MakeMove(const DungeonMove operation, const std::vector<int>& destinationIds)
     {
+        // Guard against an empty destinationIds vector
+        try
+        {
+            if (destinationIds.empty())
+            {
+                throw std::logic_error("destinationIds is empty.");
+            }
+        }
+        catch (const std::logic_error&)
```

Comment:
Laudable to check that it's empty and throw an error, but doing local recovery with a catch is not proper.  throw -> catch is meant to transfer control flow to a non-local context.

Don't do the try -catch here. Leave the throw, and handle that in e.g. HuntTheWumpus.cpp or a unit test.